xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-ui/src/main/resources/XWiki/XWikiUserSheet.xml

```diff
@@ -46,38 +46,20 @@
 ## Setting categories
 #########################
 #set($categories = [])
-#set($discard = $categories.add({'id':'profile', 'sheet':'XWiki.XWikiUserProfileSheet', 'glyphicon': 'user'}))
-#set($isMyProfile = ($services.model.resolveDocument($xcontext.user) == $doc.documentReference))
-#if($isMyProfile || $hasAdmin)
-  #set($discard = $categories.add({'id':'preferences', 'sheet':'XWiki.XWikiUserPreferencesSheet', 'glyphicon': 'wrench'}))
-#end
-#set($discard = $categories.add({'id':'groups', 'sheet':'XWiki.XWikiUserMembershipSheet', 'glyphicon': 'group'}))
-## TODO: add APIs to be able to display users watchlists to admins
-#if($isMyProfile &amp;&amp; $hasWatch)
-  #set($discard = $categories.add({'id':'watchlist', 'sheet':'XWiki.XWikiUserWatchListSheet', 'glyphicon': 'eye'}))
-#end
-#if($isMyProfile)
-  #if ("$!services.notification.watch" != '' || $hasWatch)
-    #set($discard = $categories.add({'id':'network', 'sheet':'XWiki.XWikiUserNetworkSheet', 'glyphicon': 'world'}))
-  #end
-  ## TODO: add an enhancement system instead
-  #set($notificationsSheet = $services.model.createDocumentReference('', ['XWiki', 'Notifications', 'Code'], 'XWikiUserNotificationsSheet'))
-  #if ($xwiki.exists($notificationsSheet))
-    #set($discard = $categories.add({'id':'notifications', 'sheet':'XWiki.Notifications.Code.XWikiUserNotificationsSheet', 'glyphicon': 'bell'}))
+
+## load the user menu from the 'org.xwiki.plaftorm.user.profile.menu.content' UIXP.
+#foreach ($uix in $services.uix.getExtensions('org.xwiki.plaftorm.user.profile.menu.content', {
+  'sortByCustomOrder': 'org.xwiki.platform.user.profile.entry.userprofilesheet,org.xwiki.platform.user.profile.entry.userpreferencessheet,org.xwiki.platform.user.profile.entry.usermembershipsheet,org.xwiki.platform.user.profile.entry.userwatchlistsheet,org.xwiki.platform.user.profile.entry.usernetworksheet,org.xwiki.platform.user.profile.entry.usernotificationsheet,org.xwiki.platform.user.profile.entry.userwikisheet,org.xwiki.platform.user.profile.entry.userdashboardsheet'
+}))
+  #if($uix.parameters.get('display') == 'true')
+    #if ($uix.parameters.containsKey('name'))
+        ## the name value is derived by default but can be overloaded with the name key.
+        #set($discard = $categories.add({'id': $uix.parameters.get('id'), 'name': $uix.parameters.get('name'), 'sheet': $uix.parameters.get('sheet'), 'glyphicon': $uix.parameters.get('glyphicon')}))
```

Comment:
Wouldn't it work to just do `$categories.add($uix.parameters)`? AFAICS it's already a dictionary so it should work the same way no? That would allow to remove the if above entirely too I guess.